### DISCUSSION TITLE
Max Number of Books?

### DISCUSSION THREAD

**Discussion by IamMaris** [openaudible/openaudible#1376] [Q&A] [answered]
Is there a maximum number of books OpenAudible can hold in its library? I notice it takes a good 5 minutes to open the app while it goes through the app folders to show the information, and it wasn't taking that long before the recent update. Also it is skipping a few books it used to include, but they were most likely the older "aa" extension, so I'm not too concerned about those. It is now showing 1340 books. 

One other difference I have noticed with the update, even though I have unselected the auto download and auto convert books or mp3, every time I try to search, a popup appears to see if I want to convert. It is particularly annoying if I am searching for a title and hit return, but a popup has appeared just as I hit return so it thinks I am agreeing to download or convert. Getting it to stop once it starts converting is nearly impossible.

**Comment by openaudible** [comment]
Sounds like a slow drive -- is it a network drive? Shouldn't take 5 minutes. Are you using the latest version? Something doesn't sound right.

**Comment by IamMaris** [comment]
It is a network drive, the aax folder alone is 572 GB of books. When I open
the app to add new books I purchased for the month, instead of remembering
what was previously there, it reloads each individual aax book, each image
for the art, thumbnails, and cover, and the converted books folders. That
takes about 5-6 minutes. I don't remember it reloading everything
individually before or ever taking that long in the past, and I've always
used the network drive because of the space it takes for my books. It only
started the load lag a few months ago after an update, so maybe something
glitched at that time.

Anyway, that is why I thought maybe I reached a limit the app can handle. I
was thinking of starting a new storage area to include only 2024 books to
see how it handled loading, but I was waiting to see what your response
would be.

Thanks for any suggestions. I really love this Open audible app. It's easy
to use and does a great job downloading my books and converting the books I
select for converting. All of that is super fast. Only the start up has
started to be a problem. If that's how it is supposed to work, I'll live
with it, because overall it's worth it.

Maris

**Reply by openaudible** [reply]
We needed to add some code to periodically check the file system as well as on startup. Usually it's fast but network drives can be a problem. 

I'd recommend adding an internal drive. Would make it nearly instantaneous. 

You could also check the network and nas speeds. Maybe there is a bottleneck that could be resolved some other way.

**Comment by IamMaris** [comment]
Thanks for the suggestions, and appreciate the responses, and for creating
an app that is on my "must have" list of favorites.

Maris

**Comment by dazzyp** [comment]
Hi,
I've just come across this thread after growing frustration with the worsening speed of OpenAudible.

My situation is similar to OP - over the past 3-4 years of using OpenAudible, my library has grown to 1400 titles.
The application takes minutes to open and go through its checks.  Searching is sluggish.

My library (556gb) is on a networked server on a 1gb connection. 

The application speed is the same regardless of whether I run OA on a networked PC (over a 1gb wired network) or on a windows VM running on the same server, so network speed isn't an obvious issue. 

This same server drives also hosts large collections of other media (multiple Tb with many thousands of files) and the other media management applications i use have no issues with accessing or indexing their files. 

(n.b i use Audiobookshelf to access/play my audiobook files on android devices, and Prologue/plex to to play them on apple devices)

To me, the issue is either with OA using a DB structure that can't handle large libraries, OR, it's insistence rescanning and indexing the media drive in the _foreground_ every time the app is opened  (can't this task be moved to the background and refresh out of sight??).

My workaround is to leave OA permanently open on my VM desktop. This works, but to me is a sticking plaster on the issue.   

My annual licence is expiring and i'm in the mind to try Libation, which apparently doesn't have issues with large libraries from the user comments i've seen.  Moving away from OA would be a pain as i love it otherwise, and Libation would require me building my audible downloads from scratch and doesn't have any media management functionality for non audible titles,  but its hard to justify continued payment of an application that is slowly slipping into 'too painful to use'.. i'm very sorry to say.

Happy to try any suggestions!

**Comment by openaudible** [comment]
The issue is we periodically rescan the library directories. On a local drive it takes under a second for very large libraries. 

I will try to improve performance on a NAS by testing different directory scanning techniques. We use a pretty simple method, but off the top of my head we could use a more parallel approach.. and we also can definitely reduce the number of times we scan the disk, perhaps only doing it when requested by the user (Controls: Rescan Files menu.) 

Will move the initial rescan (and as many as possible) to a background process. 

A few options until I dig deeper:

- Run OpenAudible on your NAS using OpenAudible for Docker. https://github.com/openaudible/openaudible_docker
- Buy a cheap SSD for your desktop and use rsync or something to sync converted files to your NAS.

Appreciate the feedback and will take a look at the code.

Also, if you can tell me the number of files in your OpenAudible library and the time it takes to scan, that would be good:

Controls: Refresh Files -> INFO: 8. Refreshed file cache:18741 files. [] took 16.694 seconds to complete.

**Reply by openaudible** [reply]
Spent a day reducing the frequency of full directory scans. Also used a more optimized scanner that improved performance 5x. Stand by.

**Comment by openaudible** [comment]
Here is the first version with faster directory scanning. If no reports of issues, will be rolled into future versions > 4.6.7
https://openaudible.org/beta
Let me know!

**Reply by dazzyp** [reply]
Hi OA Team, 
Thanks kindly for looking into my issue. 

I'll happily try that beta tonight and report back. 

I also wasn't aware of a docker version so will also in parallel try that in my unraid server . 

Btw I did the refresh command you asked for but can't find the output report with 'seconds to complete'. 

Nothing in the tmp/diagnostics file I can see

**Reply by openaudible** [reply]
All logs go in the File: Log Window menu item.

**Comment by dazzyp** [comment]
Hi. this the output i got on 4.6.3 . I ran it several times and got a similar result.

Refreshed file cache:2846 files. [] took 21.672 seconds to complete.</message>

Although 21 seconds doesn't seem that long, the program is blocked much longer by the 'updating book file map' which takes an age.

I'll try the beta on the same desktop for a direct comparison...

Bare with..

Thanks again

**Comment by dazzyp** [comment]
Hi again,
After installing 4.6.7.3.beta:

 Refreshed file cache:2828 files. [refreshFiles called] took 7.121 seconds to complete.</message>
 
 The 'updating book file map' section seemed faster too.
 
 In app searches don't feel much different, but i'm not sure if that was would be affected by the changes you've made.

Looks really promising - i'll check again tomorrow.

I've yet to try out the suggestion of the docker container version. That would run the app on the server fully, whereas the desktop VM (which the images runs on its own dedicated nvme drive) has is limited to only 8 core/16gb ram

Thanks for great work!

darren

**Reply by openaudible** [reply]
Good to hear. For app searches, you mean typing text into the search bar at the top left?

**Reply by dazzyp** [reply]
Hi OA Admin,
Thanks kindly again for all your work and for supporting my issue directly.

As a ways of saying thanks i've happily bought a lifetime licence.

Keep up the great work!

Cheers
Darren

**Comment by openaudible** [comment, marked as answer]
We've sped it up as much as we can, about 3x faster. 
If that doesn't work for you, try running locally on your NAS using Docker. https://openaudible.org/download
